Jenkinsfile

```
@@ -0,0 +1,118 @@
+#!groovy
+library 'pipeline-library'
+
+// Variables we can change
+def nodeVersion = '8.9.1' // NOTE that changing this requires we set up the desired version on jenkins master first!
+def sdkBranch = '7_5_X'
+
+if ('master'.equals(env.BRANCH_NAME)) {
+	// Keep last 10 builds, only run once a day
+	properties(
+		[
+			[
+				$class: 'BuildDiscarderProperty',
+				strategy: [$class: 'LogRotator', numToKeepStr: '10']
+			],
+			pipelineTriggers([cron('@midnight')]),
+		]
+	)
+}
+
+def unitTests(hostOS, guestOS, nodeVersion, sdkBranch) {
+	return {
+		def labels = ""
+		
+		if ('MacOS'.equals(hostOS)) {
+			labels += "osx"
+		} else if ('Windows'.equals(hostOS)) {
+			labels += "windows"
+		}
+		
+		if ('iOS'.equals(guestOS)) {
+			labels += ' && xcode-9'
+		} else if ('Android'.equals(guestOS)) {
+			labels += ' && android-sdk'
+		} else if ('Windows'.equals(guestOS)) {
+			labels += ' && hyper-v'
+		}
+		node(labels) {
+			unstash 'sources'
+			unstash "${hostOS}-${guestOS}"
+
+			nodejs(nodeJSInstallationName: "node ${nodeVersion}") {
+				ensureNPM('latest')
+				sh 'npm ci'
```

So I have created a "universal" pipeline command named `command` to run `sh` on unix, and `bat` on windows nodes. So long as the underlying script value is the same you can use that to automatically have it do the right thing depending on the os of the underlying node here.

So you should be able to do just:
```suggestion
				command 'npm ci'
```

It's particularly helpful when the commands you're running are npm/node scripts, but obviously if you're doing OS utils like rm/mv/cp it'd fall apart (I don't try and do any magic to change the commands to equivalents). It's stupid-simple right now: https://github.com/appcelerator/pipeline-library/blob/master/vars/command.groovy